src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImpl.java

```diff
@@ -124,4 +147,119 @@ public void removePaymentAccountAndUserAccountMaps(EntityManager em, List<UserAc
         });
 
     }
-}
\ No newline at end of file
+
+    public UpdatePbaStatusResponse updatePaymentAccountsForAnOrganisation(
+            List<PbaRequest> pbaRequestList, String orgId) {
+
+        Set<String> pbasFromRequest = new HashSet<>();
+
+        //Get PBAs from request
+        pbaRequestList.forEach(pba -> pbasFromRequest.add(pba.getPbaNumber()));
+
+        //Generate Invalid PBA Responses from Request
+        List<PbaUpdateStatusResponse> invalidPbaResponses =
+                getInvalidPbaRequests(pbaRequestList, pbasFromRequest, orgId);
+
+        //Remove invalid PBAs from PBAs to update
+        invalidPbaResponses.forEach(pba -> pbasFromRequest.remove(pba.getPbaNumber()));
+
+        //Update valid PBAs, returns any remaining invalid PBAs
+        if (isNotEmpty(pbasFromRequest)) {
+            invalidPbaResponses = acceptOrRejectPbas(pbasFromRequest, pbaRequestList, invalidPbaResponses);
+        }
+
+        return generateUpdatePbaResponse(invalidPbaResponses, pbasFromRequest);
+    }
+
+    private List<PbaUpdateStatusResponse> getInvalidPbaRequests(
+            List<PbaRequest> pbaRequestList, Set<String> pbasFromRequest, String orgId) {
+
+        List<PbaUpdateStatusResponse> invalidPbaRequests = new ArrayList<>();
+
+        pbaRequestList.forEach(pba -> {
+            //check pba is present
+            if (isNullOrEmpty(pba.getPbaNumber())) {
+                invalidPbaRequests.add(new PbaUpdateStatusResponse("", ERROR_MSG_PBA_MISSING));
+                //check status is present
+            } else if (isNullOrEmpty(pba.getStatus())) {
+                invalidPbaRequests.add(new PbaUpdateStatusResponse(pba.getPbaNumber(), ERROR_MSG_STATUS_MISSING));
+                pbasFromRequest.remove(pba.getPbaNumber());
+                //check status is valid
+            } else if (!ACCEPTED.name().equalsIgnoreCase(pba.getStatus())
+                    && !REJECTED.name().equalsIgnoreCase(pba.getStatus())) {
+                invalidPbaRequests.add(new PbaUpdateStatusResponse(pba.getPbaNumber(), ERROR_MSG_STATUS_INVALID));
+                pbasFromRequest.remove(pba.getPbaNumber());
+            }
+        });
+
+        //Check individual PBA numbers are valid and exist within given organisation
+        invalidPbaRequests.addAll(getInvalidPbas(pbasFromRequest, orgId));
+
+        return invalidPbaRequests;
+    }
+
+    public List<PbaUpdateStatusResponse> getInvalidPbas(Set<String> pbasFromRequest, String orgId) {
+        List<PbaUpdateStatusResponse> pbaUpdateStatusResponses = new ArrayList<>();
+
+        pbasFromRequest.forEach(pba -> {
+            //check PBA has a valid format
+            if (isPbaInvalid(pba)) {
+                pbaUpdateStatusResponses.add(new PbaUpdateStatusResponse(pba, ERROR_MSG_PBA_INVALID_FORMAT));
+            } else {
+                Optional<PaymentAccount> paymentAccount = paymentAccountRepository.findByPbaNumber(pba);

```

Comment:
please reduce call to db in for loop , use IN clause rather

Reply:
Addressed here: https://github.com/hmcts/rd-professional-api/pull/1096/commits/db27d8f70138e98e2f119e5d6c7f0f350cba04fa